### PR TITLE
refactor: reduce casting

### PR DESCRIPTION
**Pull request by iamacook**
This reduces our reliance on type casting.

### CODE REVIEW

File: src/domain/safe/entities/transfer.entity.ts [hunk 1]
```diff
@@ -7,14 +7,30 @@ export interface Transfer {
...
+
 export interface ERC20Transfer extends Transfer {
   tokenAddress: string;
   value: string;
 }
 
 export function isERC20Transfer(transfer: Transfer): transfer is ERC20Transfer {
-  const erc20Transfer = transfer as ERC20Transfer;
-  return erc20Transfer.tokenAddress != null && erc20Transfer.value != null;
+  const hasValue = 'value' in transfer && transfer.value != null;
```
Comment (hectorgomezv) [inline src/domain/safe/entities/transfer.entity.ts:32]: I'm a bit confused about this line. Isn't `hasValue` already declared as a function above?

File: src/domain/safe/entities/transfer.entity.ts [hunk 2]
```diff
@@ -25,8 +41,8 @@ export interface ERC721Transfer extends Transfer {
 export function isERC721Transfer(
   transfer: Transfer,
 ): transfer is ERC721Transfer {
-  const erc721Transfer = transfer as ERC721Transfer;
-  return erc721Transfer.tokenAddress != null && erc721Transfer.tokenId != null;
+  const hasTokenId = 'tokenId' in transfer && transfer.tokenId != null;
```
Comment (hectorgomezv) [inline src/domain/safe/entities/transfer.entity.ts:44]: Following the logic on `isERC20Transfer` we could also extract `hasTokenId` to a function to have a clearer implementation, wdyt?

File: src/datasources/errors/http-error-factory.ts [hunk 3]
```diff
@@ -32,6 +32,8 @@ export class HttpErrorFactory {
 function isNetworkResponseError(
   error: NetworkError | Error,
 ): error is NetworkResponseError {
-  const responseError = error as NetworkResponseError;
-  return responseError.status >= 400 && responseError.status < 600;
+  if ('status' in error) {
```
Comment (hectorgomezv) [inline src/datasources/errors/http-error-factory.ts:35]: I think we could be simplyfied even further:
```
return 'status' in error && error.status >= 400 && error.status < 600;
```

**Review by hectorgomezv — COMMENTED** [review]
Great job! I've just added some nitpick/minor comments 🙂

File: src/domain/safe/entities/transfer.entity.ts (hunk 2 above)
Comment (iamacook) [reply src/domain/safe/entities/transfer.entity.ts]: Good idea. I've extracted the logic accordingly in 7cc62fa26e597a277f5adc5a0bda4ed3cba0e0fc.

File: src/domain/safe/entities/transfer.entity.ts (hunk 1 above)
Comment (iamacook) [reply src/domain/safe/entities/transfer.entity.ts]: It is, I missed this one. I've remove it in 7cc62fa26e597a277f5adc5a0bda4ed3cba0e0fc.

File: src/datasources/errors/http-error-factory.ts (hunk 3 above)
Comment (iamacook) [reply src/datasources/errors/http-error-factory.ts]: Great suggestion. I've change it in 7cc62fa26e597a277f5adc5a0bda4ed3cba0e0fc.

**Review by hectorgomezv — APPROVED** [review]
🚀🚀🚀

**Pull request merged**